test/parallel/test-runner-mock-timers.js

```diff
@@ -479,6 +479,59 @@ describe('Mock Timers Test Suite', () => {
             code: 'ERR_INVALID_ARG_TYPE',
           });
         });
+
+        // Test for https://github.com/nodejs/node/issues/50365
+        it('should not affect other timers when aborting', async (t) => {
+          const f1 = t.mock.fn();
+          const f2 = t.mock.fn();
+          t.mock.timers.enable({ apis: ['setTimeout'] });
+          const ac = new AbortController();
+
+          // id 1 & pos 1 in priority queue
+          nodeTimersPromises.setTimeout(100, undefined, { signal: ac.signal }).then(f1, f1);
+          // id 2 & pos 1 in priority queue (id 1 is moved to pos 2)
+          nodeTimersPromises.setTimeout(50).then(f2, f2);
+
+          ac.abort(); // BUG: will remove timer at pos 1 not timer with id 1!
+
+          t.mock.timers.runAll();
+          await nodeTimersPromises.setImmediate(); // let promises settle
+
+          // First setTimeout is aborted
+          assert.strictEqual(f1.mock.callCount(), 1);
+          assert.strictEqual(f1.mock.calls[0].arguments[0].code, 'ABORT_ERR');
+
+          // Second setTimeout should resolve, but never settles, because it was eronously removed by ac.abort()
+          assert.strictEqual(f2.mock.callCount(), 1);
+        });
+
+        // Test for https://github.com/nodejs/node/issues/50365
+        it('should not affect other timers when aborted after triggering', async (t) => {
+          const f1 = t.mock.fn();
+          const f2 = t.mock.fn();
+          t.mock.timers.enable({ apis: ['setTimeout'] });
+          const ac = new AbortController();
+
+          // id 1 & pos 1 in priority queue
+          nodeTimersPromises.setTimeout(50, true, { signal: ac.signal }).then(f1, f1);
+          // id 2 & pos 2 in priority queue
+          nodeTimersPromises.setTimeout(100).then(f2, f2);
+
+          // First setTimeout resolves
+          t.mock.timers.tick(50);
```

Comment:
I think we should use platform timeout value from common in all constant timeout and internal values.

Reply:
Since the timer is mocked, does it really matter here?